File: java/src/main/java/com/tomoarrow/idv/bff/java/service/StateService.java
package com.tomoarrow.idv.bff.java.service;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;
import org.springframework.stereotype.Service;

@Service
public class StateService {
    private final Map<String, Object> values = new ConcurrentHashMap<>();

    public void set(String key, Object value) {
        values.put(key, value);
    }

    public Object get(String key) {
        return values.get(key);
    }

    public void clear() {
        values.clear();
    }
}

File: kotlin/src/main/kotlin/com/tomoarrow/idv/bff/service/StateService.kt
package com.tomoarrow.idv.bff.service

import org.springframework.stereotype.Service
import java.util.concurrent.ConcurrentHashMap

/**
 * In-memory state storage for access tokens and other session data.
 * Equivalent to the TypeScript StateService singleton.
 */
@Service
class StateService {
    private val state = ConcurrentHashMap<String, Any>()

    fun set(key: String, value: Any) {
        state[key] = value
    }

    fun get(key: String): Any? = state[key]

    fun has(key: String): Boolean = state.containsKey(key)

    fun delete(key: String): Boolean = state.remove(key) != null

    fun getAll(): Map<String, Any> = state.toMap()

    fun clear() = state.clear()
}
